### PR TITLE
Auto save

### LINKED ISSUE #11: Build in auto-save feature
Add save features to buildFileManifest, MD step, classifier step

Include save date in file name

### LINKED ISSUE #22: loadData()
Add a loadData() function that takes a file and loads it into the proper object type.  

Within any function with an outfile parameter, check if outfile already exists at start, ask to load, if yes, load file and abort function.

### PR DESCRIPTION
**Pull request by Zracano**
Added save and load features for functions creating and modifying data frames. 
Closes #22 , Closes #11

### CODE REVIEW

File: R/applyPredictions.R [hunk 1]
```diff
@@ -15,6 +15,13 @@
 #' alldata <- applyPredictions(animals,empty,classfile,pred,counts = FALSE)
 #' }
 applyPredictions <- function(animals, empty, classfile, pred, outfile = NULL, counts = FALSE) {
+  if (!is.na(outfile) && file.exists(outfile)) {
```
Comment (tkswanson) [inline R/applyPredictions.R:18]: change default parameter outfile = NA

File: R/imagesFromVideos.R [hunk 2]
```diff
@@ -18,21 +18,20 @@
...
 #' }
-imagesFromVideos <- function(files, outdir = tempfile(), outfile = NULL, format = "jpg", fps = NULL, frames = NULL, parallel = FALSE, nproc = 1) {
-  if (!is(files, "data.frame")) {
-    stop("Error: 'mdresults' must be Data Frame.")
-  }
-  if (outdir != "" & !dir.exists(outdir)) {
-    if (!dir.create(outdir, recursive = TRUE)) {
-      stop("Output directory invalid.\n")
+imagesFromVideos <- function(files, outdir = tempfile(), outfile = NA, format = "jpg", fps = NULL, frames = NULL, parallel = FALSE, nproc = 1) {
+  if (!is.na(outfile) && file.exists(outfile)) {
```
Comment (tkswanson) [inline R/imagesFromVideos.R:22]: this exists in 2+ places, lets make it a function

File: R/setLocation.R [hunk 3]
```diff
@@ -15,7 +15,14 @@
 #' \dontrun{
 #' setLocation(files, basedir)
 #' }
-setLocation <- function(files, basedir, adjust = 0, rename = TRUE, region = NA, site = NA, camera = NA) {
+setLocation <- function(files, basedir, outfile = NA, adjust = 0, rename = TRUE, region = NA, site = NA, camera = NA) {
```
Comment (tkswanson) [inline R/setLocation.R:18]: added param to roxygen ?

File: examples/FullWorkflow_animl.R [hunk 4]
```diff
@@ -76,7 +76,7 @@ alldata <- applyPredictions(animals,empty,"/mnt/machinelearning/Models/Southwest
 
 # Classify sequences
 #mdanimals <- classifySequence(mdanimals,pred,classes,18,maxdiff=60)
-alldata <- poolCrops(alldata)
+alldata <- poolCrops(alldata, outfile = predresults)
```
Comment (tkswanson) [inline examples/FullWorkflow_animl.R:79]: I'll probable change these but thats ok

**Review by tkswanson — CHANGES_REQUESTED** [review]
Since we are using that overwrite check multiple times, lets make it a function so that we only have to modify it once in a single file going forward.

File: R/imagesFromVideos.R (hunk 2 above)
Comment (Zracano) [reply R/imagesFromVideos.R:22]: Would it be okay to add it to the loadData function? So it checks for the files existence and prompt the user if it exists, or make it a separate function within the same file?

**Review by tkswanson — APPROVED** [review]

**Pull request merged**